### ISSUE TITLE
Abide validation problem in modal

### ISSUE THREAD

**Issue by stavros-zavrakas** [zurb/foundation-sites#6740]
Hi,

I have a page that contains a form (let's call it form_body) and a modal with another form (let's call it form_modal). Both of them they have two fields that are required. On form_body form the validations works without problems. On the other hand form_modal's validation is triggered only when I am trying to submit the form. The keyup and all the other validations are not triggered at all. I tried different workarounds like re-initalising the foundation, doing something like that $(document).foundation('abide','events'), moving around the modal DOM etc without success. I found that also other people have similar problem. Is there a possibility to take a look into that or can you suggest a fast patch that can be acceptable? Here is the specific example that I created:

```
<!doctype html>
<html>

  <head>
    <meta charset="utf-8">
    <meta http-equiv="x-ua-compatible" content="ie=edge">
    <title>Foundation tests</title>
    <meta name="description" content="">
    <meta name="viewport" content="width=device-width, initial-scale=1">

    <link rel="stylesheet" href="bower_components/foundation/css/foundation.css">

  </head>

  <body>
    <div class="modalWrapper">
      <div id="firstModal" class="reveal-modal" data-reveal aria-labelledby="firstModalTitle" aria-hidden="true" role="dialog">
        <h2 id="firstModalTitle">This is a modal.</h2>

        <form data-abide="ajax">
          <div class="name-field">
            <label>Your first name name <small>required</small>
              <input type="text" required pattern="[a-zA-Z]+">
            </label>
            <small class="error">File is required and must be a string.</small>
          </div>
          <div class="surname-field">
            <label>Your last name name <small>required</small>
              <input type="text" required pattern="[a-zA-Z]+">
            </label>
            <small class="error">File is required and must be a string.</small>
          </div>
          <button type="submit">Submit</button>
        </form>

        <a class="close-reveal-modal" aria-label="Close">&#215;</a>
      </div>
    </div>

    <div class="row">
      <div class="large-6 columns">
        <h2>Abide validation example</h2>
      </div>
      <div class="large-6 columns">
        <h2>Abide validation modal</h2>
      </div>
    </div>

    <div class="row">
      <div class="large-6 columns">
        <form data-abide="ajax">
          <div class="name-field">
            <label>Your name <small>required</small>
              <input type="text" required pattern="[a-zA-Z]+">
            </label>
            <small class="error">Name is required and must be a string.</small>
          </div>
          <div class="email-field">
            <label>Email <small>required</small>
              <input type="email" required>
            </label>
            <small class="error">An email address is required.</small>
          </div>
          <button type="submit">Submit</button>
        </form>
      </div>
      <div class="large-6 columns">
        <a href="#" data-reveal-id="firstModal" class="radius button">Form in a modal</a>
      </div>
    </div>


    <script src="bower_components/jquery/dist/jquery.js"></script>
    <script src="bower_components/foundation/js/foundation.js"></script>
    <script>
      $(document).foundation();
    </script>
  </body>

</html>
```

and the bower.json file

```
{
  "name": "handlebars test",
  "version": "0.0.0",
  "authors": [
    "Stavros Zavrakas <szavrakas@cambridge.org>"
  ],
  "license": "MIT",
  "ignore": [
    "**/.*",
    "node_modules",
    "bower_components",
    "test",
    "tests"
  ],
  "dependencies": {
    "foundation": "~5.5.2",
    "jquery": "~2.1.4"
  }
}
```

**Comment by stavros-zavrakas** [comment]
Here you can find a working version of the code above in jsbin:
https://jsbin.com/dijetetuhu/edit?html,js,output

**Comment by rafibomb** [comment]
Have you tried reflow?  $(document).foundation('abide', 'reflow');

**Comment by stavros-zavrakas** [comment]
I tried it and it is fixing that specific problem. The thing is that I have attached some other custom events on the form and after the $(document).foundation('abide', 'reflow'); it removes the custom binding and I have to rebind again.

For example I have this custom binding and after the reflow I have to rebind it again:
$uploadForm.on('valid.fndtn.abide', function (e) {
      e.preventDefault();
     ......
});

**Comment by Nauzer** [comment]
Hey Stavros,

I'm experiencing the same problem at the moment. Can't seem to find any error related to it either... Did you find a solution for this?

Cheers!

Mark

**Comment by rafibomb** [comment]
Any ideas on how to solve this for your use case?

**Comment by dkperson** [comment]
Hey guys, I think i traced the root cause of this issue too (https://github.com/zurb/foundation/pull/6407). I had been experiencing this problem for a while since the 5.2 release and I went through all the updates to Abide JS since 5.1.2.
I found that the following change in: (https://github.com/zurb/foundation/commit/ba8c41fa877715ae3de4e8ba4d36ce4a408b4dae#diff-0fecfcb7f50be59e4f71ef75a2d8be41)
breaks the form input errors in the modal.
I believe this is because the fields are marked as hidden because the modal is hidden, and since the new change does not validate hidden input fields, the errors don't pop up.

I have also modified the Fiddle stavros-zavrakas posted above, to use the live-validate option with a fast refresh rate.
https://jsbin.com/weyizaniza/edit?html,js,output

If you click on a input field and then click out of it, or onto another in the first form, you will see that the error instantly pops ups. While in the modal form, the input fields do not show any errors because they are not being validated.
